tutorials/tour/basics.md

```diff
@@ -0,0 +1,207 @@
+---
+layout: tutorial
+title: Basics
+
+disqus: true
+
+tutorial: scala-tour
+num: 2
+next-page: unified-types
+previous-page: tour-of-scala
+---
+
+In this page, we will cover basics of Scala.
+
+## Variables
+
+You can define variables with `var` keyword.
+
+```
+var x = 1 + 1
+x += 1
+```
+
+Often, you want your variables to be immutable. You can use `val` keyword in that case.
+
+```
+val x = 1 + 1
+x += 1 // This does not compile because you declared the variable with "val" keyword
+```
+
+Type of variables can be inferred, but you can also explicitly state type like below.
+
+```
+val x: Int = 1 + 1
+```
+
+## Functions
+
+You can define functions like below.
+
+```
+(name: String) => println("Hello, " + name + "!")
+```
+
+You can also assign functions to variables.
+
+```
+val greet = (name: String) => println("Hello, " + name + "!")
+greet("Scala") // Hello, Scala!
+```
+
+We will cover functions in depth [later](anonymous-function-syntax.md).
+
+## Classes
+
+You can define classes with `class` keyword.
+
+```
+class Point(var x: Int, var y: Int) {
+  def move(dx: Int, dy: Int): Unit = {
+    x = x + dx
+    y = y + dy
+  }
+
+  override def toString: String = "(" + x + ", " + y + ")"
+}
+```
+
+You can instantiate classes with `new` keyword.
+
+```
+val point = new Point(1, 2)
+println(point) // (1, 2)
+```
+
+We will cover classes in depth [later](classes.md).
+
+## Case Classes
+
+By default, classes are compared by reference. If you want to compare them by values, you need to override `equals` and `hashCode` methods.
+
+```
+class Point(val x: Int, val y: Int) {
+  override def equals(o: Any) = o match {
+    case that: Point => x == that.x && y == that.y
+    case _ => false
+  }
+  override def hashCode = (x.hashCode << 16) + y.hashCode
+}
+```
+
+While it works, it's tedious to define those methods for all classes you want to compare by value.
+
+Thankfully, Scala has another type of class called `case class` that's compared by value by default.
+
+```
+case class Point(x: Int, y: Int)
+```
+
+You can instantiate case classes without `new` keyword.
+
+```
+val point = Point(1, 2) // no "new" here
+val anotherPoint = Point(1, 2)
+val yetAnotherPoint = Point(2, 2)
+println(point == anotherPoint) // true
+println(point == yetAnotherPoint) // false
+```
+
+Case classes are immutable by default, but it also provides `copy` method so that you can easily create another instance of the class while reusing the values from exiting instances.
```

Comment:
It sounds better, but I think "modify any of the values defined in a case class" also sounds like it will mutate values, which is not what's happening. I cannot think of a simple way of wording what's actually happening, though. I started to think maybe `copy` is not worth mentioning in this page.